File: tests/functional/Bots/Python/Consumers/CodeFirst/WaterfallHostBot/skills/echo_skill.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

from enum import Enum
from botbuilder.schema import Activity
from skills.skill_definition import SkillDefinition


class EchoSkill(SkillDefinition):
    class SkillAction(str, Enum):
        MESSAGE = "Message"

    def get_actions(self):
        return self.SkillAction

    def create_begin_activity(self, action_id: str):
        if action_id not in self.SkillAction:
            raise Exception(f'Unable to create begin activity for "${action_id}".')

        # We only support one activity for Echo so no further checks are needed
        activity = Activity.create_message_activity()
        activity.name = self.SkillAction.MESSAGE.value
        activity.text = "Begin the Echo Skill"

        return activity

File: tests/functional/Bots/DotNet/Consumers/CodeFirst/WaterfallHostBot/Skills/EchoSkill.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using Microsoft.Bot.Schema;

namespace Microsoft.BotFrameworkFunctionalTests.WaterfallHostBot.Skills
{
    public class EchoSkill : SkillDefinition
    {
        private enum SkillAction
        {
            Message
        }

        public override IList<string> GetActions()
        {
            return new List<string> { SkillAction.Message.ToString() };
        }

        public override Activity CreateBeginActivity(string actionId)
        {
            if (!Enum.TryParse<SkillAction>(actionId, true, out _))
            {
                throw new InvalidOperationException($"Unable to create begin activity for \"{actionId}\".");
            }

            // We only support one activity for Echo so no further checks are needed
            return new Activity(ActivityTypes.Message)
            {
                Name = SkillAction.Message.ToString(),
                Text = "Begin the Echo Skill."
            };
        }
    }
}
